api_types.ts

```diff
@@ -689,6 +689,13 @@ export interface FetchResponse<T extends any = any> {
    * to disable any parsing. Note however that this will result in the body being a NodeJS Buffer.
    */
   body?: T;
+  /**
+   * The compression used for body in the raw HTTP response (e.g. gzip, deflate).
+   * 
+   * Note that the response {@link FetchResponse.body} is automatically decompressed if this is gzip or deflate,
+   * otherwise, you will need to handle decompression, including cases where multiple compressions are used.
+   */
+  bodyCompression?: string;
```

Comment:
This seems reasonable as an interface (would probably call it simply `compression` to be concise) but are we actually going to make use of it in solving the OOMs? Seems like we should start by just moving the existing automatic decompression into the lambda, unless you had a specific plan to also use this within specific packs?

Reply:
Need this field so we know which responses we need to decompress in lambda, unless there is a better way?

Reply:
@patrick-codaio mentioned that it would also be helpful for developers to figure out why lambda is OOMing.

Reply:
Ah, my bad. I thought you were adding this to the _request_ interface, not the _response_ (and so expecting a bunch of packs to update to start setting this). Nevermind, sorry for the confusion.

Yeah, you do need a bit somewhere to tell the lambda if and how a response was compressed. But I thought at the http level this is communicated in the headers, which are already part of the fetch response. So would just try to use the `Content-Encoding` header or whatever, which is presumably what you'd look at to set this bit in the first place.

Reply:
Yeah this is basically `headers['Content-Encoding'] === 'gzip' && !config.decompress`, I don't see a good way to pass in runtime config into packs runtime so added this field.

Reply:
Ah, these are all runtime config settings that are passed into the lambda that you can key new behaviors off of:

https://github.com/coda/coda/blob/4f93caa98bf21eed78ddbcb5cdfceb2fdd775441/modules/common-internal/runtime-config/settings.ts#L1347-L1356

So would add a new property for this here, and then pass that value into the node fetcher to turn off automatic decompression and then look at the same value in `this._packRuntimeOptions` in the fetcher proxy to determine whether should decompress based on the response header.

Let me know if I'm still misunderstanding though.

There's a separate thing we have to address before rolling this out, which is that in order to know whether it's appropriate to do the new behavior in the node fetcher, we need to know if the pack version is up to date enough to have the new logic in the lambda. So need to also look at something like https://github.com/coda/coda/blob/001fdea61d6284d24eb9dd4f17d85cc40a5597b0/modules/common-internal/types/packs/index.ts#L65 in addition to the runtime config to enable the new behavior. (`sdkVersionMatchesFeature()` is used for a similar need when the actual SDK is changing, but in this case we only need to change the lambda so we need only look at the lambda version or whatever). But you can build the new behavior based only on the runtime config and get PRs merged and we can sort out this additional check before turning on the runtime config.

Reply:
I think that would work, thanks for the pointers!